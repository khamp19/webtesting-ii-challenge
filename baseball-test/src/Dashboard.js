import React, { Component } from 'react';
import Display from './Display';

class Dashboard extends Component {
  constructor(){
    super();
    this.state = {
      strikes: 0,
      balls: 0,
      fouls: 0,
      hits: 0,
      outs: 0,
    }
  }

  incStrikes = () => {
    let strikesUp = this.state.strikes;
    if (strikesUp === 2){
      this.incOuts();
      this.setState({ strikes: 0 });
    } else {
      console.log('strike!');
      strikesUp++;
      this.setState({ strikes: strikesUp })
    }
  }

  incFouls = () => {
    console.log('foul ball!');
    let fouls = this.state.fouls;
    let strikes = this.state.strikes;
    if(fouls >= 2){
      console.log('no strike');
      fouls++;
      this.setState({ fouls: fouls });
    } else {
      fouls++;
      strikes++;
      this.setState({
        fouls: fouls,
        strikes: strikes,
      })
    }
  }

  incBalls = () => {
    let balls = this.state.balls;
    if(balls === 3){
      console.log('walk');
      this.incHits();
      this.setState({
        balls: 0,
        strikes: 0,
        fouls: 0,
      })
    } else {
      console.log('bad pitch');
      balls++;
      this.setState({ balls: balls });
    }
  }

  incHits = () => {
    console.log('home run!');
    let hits = this.state.hits;
    hits++;
    this.setState({
      strikes: 0,
      balls: 0,
      fouls: 0,
      hits: hits,
    })
  }

  incOuts = () => {
    console.log(`you're out!`);
    let outs = this.state.outs;
    if(outs === 2){
      alert('end of inning');
      this.setState({
        strikes: 0,
        balls: 0,
        fouls: 0,
        hits: 0,
        outs: 0,
      })
    } else {
      outs++;
      this.setState({
        strikes: 0,
        balls: 0,
        fouls: 0,
        outs: outs,
      })
    }
  }

  clear = () => {
    console.log('clear board');
    this.setState({
      strikes: 0,
      balls: 0,
      fouls: 0,
      hits: 0,
      outs: 0,
    })
  }

  render(){
    return(
      <div>
        <h2>hello from Dashboard</h2>
        <Display {...this.state}/>
        <div>
          <button className='strike' onClick={this.incStrikes}>Strike</button>
          <button className='ball' onClick={this.incBalls}>Ball</button>
          <button className='foul' onClick={this.incFouls}>Foul</button>
          <button className='hit' onClick={this.incHits}>Hit</button>
        </div>
        <button onClick={this.clear}>Clear</button>
      </div>
    )
  }
}

export default Dashboard;