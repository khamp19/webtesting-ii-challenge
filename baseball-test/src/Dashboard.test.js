import React from 'react';
import ReactDOM from 'react-dom';
import Dashboard from './Dashboard';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dashboard />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should start with state at 0', () => {
  const wrapper = shallow(<Dashboard strikes={0} balls={0} hits={0} fouls={0} outs={0} />)
  expect(wrapper).toMatchSnapshot();
});

it('should change the strike state', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.strike'); 

  expect(instance.state.strikes).toBe(0);
  button.simulate('click')
  expect(instance.state.strikes).toBe(1);
  button.simulate('click')
  expect(instance.state.strikes).toBe(2);
});

it('should reset the strikes, balls, and fouls with 3 strikes', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.strike');

  expect(instance.state.strikes).toBe(0);
  button.simulate('click');
  expect(instance.state.strikes).toBe(1);
  button.simulate('click');
  expect(instance.state.strikes).toBe(2);
  button.simulate('click');
  expect(instance.state.strikes).toBe(0);
  expect(instance.state.balls).toBe(0);
  expect(instance.state.fouls).toBe(0);
});

it('should increase the outs with 3 strikes', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.strike');

  expect(instance.state.strikes).toBe(0);
  button.simulate('click');
  expect(instance.state.strikes).toBe(1);
  button.simulate('click');
  expect(instance.state.strikes).toBe(2);
  button.simulate('click');
  expect(instance.state.strikes).toBe(0);
  expect(instance.state.outs).toBe(1);
});

it('should change the ball state', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.ball');

  expect(instance.state.balls).toBe(0);
  button.simulate('click')
  expect(instance.state.balls).toBe(1);
  button.simulate('click')
  expect(instance.state.balls).toBe(2);
});

it('should reset the fouls, strikes, and balls with 4 balls', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.ball');

  expect(instance.state.balls).toBe(0);
  button.simulate('click')
  expect(instance.state.balls).toBe(1);
  button.simulate('click')
  expect(instance.state.balls).toBe(2);
  button.simulate('click');
  expect(instance.state.balls).toBe(3);
  button.simulate('click');
  expect(instance.state.balls).toBe(0);
  expect(instance.state.strikes).toBe(0);
  expect(instance.state.fouls).toBe(0);
});

it('should increase the hits with 4 balls', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.ball');

  expect(instance.state.balls).toBe(0);
  button.simulate('click');
  expect(instance.state.balls).toBe(1);
  button.simulate('click');
  expect(instance.state.balls).toBe(2);
  button.simulate('click');
  expect(instance.state.balls).toBe(3);
  button.simulate('click');
  expect(instance.state.balls).toBe(0);
  expect(instance.state.hits).toBe(1);
});

it('should change the hit state', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.hit');

  expect(instance.state.hits).toBe(0);
  button.simulate('click')
  expect(instance.state.hits).toBe(1);
  button.simulate('click')
  expect(instance.state.hits).toBe(2);
});

it('a hit should reset balls, fouls, and strikes', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.hit');

  expect(instance.state.hits).toBe(0);
  button.simulate('click')
  expect(instance.state.hits).toBe(1);
  expect(instance.state.balls).toBe(0);
  expect(instance.state.strikes).toBe(0);
  expect(instance.state.fouls).toBe(0);
});

it('should change the foul state', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.foul');

  expect(instance.state.fouls).toBe(0);
  button.simulate('click')
  expect(instance.state.fouls).toBe(1);
  button.simulate('click')
  expect(instance.state.fouls).toBe(2);
});

it('fouls should increase strikes up to 2', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.foul');

  expect(instance.state.fouls).toBe(0);
  expect(instance.state.strikes).toBe(0)
  button.simulate('click')
  expect(instance.state.fouls).toBe(1);
  expect(instance.state.strikes).toBe(1)
  button.simulate('click')
  expect(instance.state.fouls).toBe(2);
  expect(instance.state.strikes).toBe(2);
});

it('fouls should not cause 3 strikes', () => {
  const wrapper = shallow(<Dashboard />)
  const instance = wrapper.instance();
  const button = wrapper.find('button.foul');

  expect(instance.state.fouls).toBe(0);
  expect(instance.state.strikes).toBe(0)
  button.simulate('click')
  expect(instance.state.fouls).toBe(1);
  expect(instance.state.strikes).toBe(1)
  button.simulate('click')
  expect(instance.state.fouls).toBe(2);
  expect(instance.state.strikes).toBe(2);
  button.simulate('click')
  expect(instance.state.fouls).toBe(3);
  expect(instance.state.strikes).toBe(2);
});

// const fouls = wrapper.find('button.foul');
// const balls = wrapper.find('button.ball');
// const hit = wrapper.find('button.hit');

// it('should reset the strikes, balls, fouls, and hits with 3 outs', () => {
//   const wrapper = shallow(<Dashboard strikes={2} />)
//   const instance = wrapper.instance();
//   const button = wrapper.find('button.strike');

//   expect(instance.state.strikes).toBe(0);
//   button.simulate('click');
//   expect(instance.state.strikes).toBe(1);
//   button.simulate('click');
//   expect(instance.state.strikes).toBe(2);
//   button.simulate('click');
//   expect(instance.state.strikes).toBe(0);
// });