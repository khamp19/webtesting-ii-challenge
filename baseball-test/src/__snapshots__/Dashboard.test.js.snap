// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start with state at 0 1`] = `ShallowWrapper {}`;
